test/src/DistanceFieldTest.cpp

```diff
@@ -1,261 +1,314 @@
+class DistanceFieldTest : public RGLTestWithParam<float>
+{
+protected:
+	rgl_node_t useRaysNode = nullptr;
+	rgl_node_t raytraceNode = nullptr;
+	rgl_node_t gaussianNoiseNode = nullptr;
+	rgl_node_t yieldPointsNode = nullptr;
+	rgl_node_t compactPointsNode = nullptr;
+	rgl_node_t transformPointsNode = nullptr;
+
+	std::vector<rgl_mat3x4f> rayTf;
+
+	std::vector<rgl_field_t> fields = {DISTANCE_F32, XYZ_F32};
+	std::vector<::Field<DISTANCE_F32>::type> outDistances;
+	std::vector<::Field<XYZ_F32>::type> outPoints;
+
+	std::unique_ptr<PointCloud> pointCloud;
+
+	void connectNodes(bool withGaussianNoiseNode)
+	{
+		ASSERT_RGL_SUCCESS(rgl_graph_node_add_child(useRaysNode, raytraceNode));
+		ASSERT_RGL_SUCCESS(rgl_graph_node_add_child(raytraceNode, compactPointsNode));
+		if (withGaussianNoiseNode) {
+			ASSERT_RGL_SUCCESS(rgl_graph_node_add_child(compactPointsNode, gaussianNoiseNode));
+			ASSERT_RGL_SUCCESS(rgl_graph_node_add_child(gaussianNoiseNode, yieldPointsNode));
+		} else {
+			ASSERT_RGL_SUCCESS(rgl_graph_node_add_child(compactPointsNode, yieldPointsNode));
+		}
+	}
+
+	void disconnectNodes(bool withGaussianNoiseNode)
+	{
+		ASSERT_RGL_SUCCESS(rgl_graph_node_remove_child(useRaysNode, raytraceNode));
+		ASSERT_RGL_SUCCESS(rgl_graph_node_remove_child(raytraceNode, compactPointsNode));
+		if (withGaussianNoiseNode) {
+			ASSERT_RGL_SUCCESS(rgl_graph_node_remove_child(compactPointsNode, gaussianNoiseNode));
+			ASSERT_RGL_SUCCESS(rgl_graph_node_remove_child(gaussianNoiseNode, yieldPointsNode));
+		} else {
+			ASSERT_RGL_SUCCESS(rgl_graph_node_remove_child(compactPointsNode, yieldPointsNode));
+		}
+	}
+
+	void prepareNodes()
+	{
+		ASSERT_RGL_SUCCESS(rgl_node_rays_from_mat3x4f(&useRaysNode, rayTf.data(), rayTf.size()));
+		ASSERT_RGL_SUCCESS(rgl_node_raytrace(&raytraceNode, nullptr, RAYTRACE_DEPTH));
+		ASSERT_RGL_SUCCESS(rgl_node_points_yield(&yieldPointsNode, fields.data(), fields.size()));
+		ASSERT_RGL_SUCCESS(rgl_node_points_compact(&compactPointsNode));
+	}
+
+	void getResults(rgl_node_t node = nullptr)
+	{
+		ASSERT_RGL_SUCCESS(rgl_graph_run(useRaysNode));
+
+		if (node == nullptr) {
+			node = yieldPointsNode;
+		}
+		pointCloud = std::make_unique<PointCloud>(PointCloud::createFromNode(node, fields));
+		ASSERT_EQ(pointCloud->getPointCount(), rayTf.size());
+		outDistances = pointCloud->getFieldValues<DISTANCE_F32>();
+		outPoints = pointCloud->getFieldValues<XYZ_F32>();
+	}
+};
+
+INSTANTIATE_TEST_CASE_P(DistanceFieldTests, DistanceFieldTest, ::testing::Values(23.1753849f, 119.9884538f, 333.3394629f));
 
 TEST_P(DistanceFieldTest, should_compute_correct_distance_for_single_object_on_ray_path)
 {
-    float cubeZDistance = GetParam();
-    const rgl_mat3x4f cubePoseTf = Mat3x4f::translation(0, 0, cubeZDistance).toRGL();
-    prepareSceneWithCube(cubePoseTf);
+	float cubeZDistance = GetParam();
+	const Mat3x4f cubePoseTf = Mat3x4f::translation(0, 0, cubeZDistance);
+	spawnCubeOnScene(nullptr, cubePoseTf);
```

Comment:
Not sure how this change is better.

Reply:
The `spawnCubeOnScene` function existed before the `prepareSceneWithCube` function. When creating the `prepareSceneWithCube` function, I did not know about the `spawnCubeOnScene` function. So I removed the unnecessary duplicate.

Reply:
I just do not like this nullptr.